Jenkinsfile

```diff
@@ -1,216 +1,64 @@
 #!groovy
 
-@Library('github.com/mozmar/jenkins-pipeline@20170214.1')
+@Library('github.com/mozmar/jenkins-pipeline@20170315.1')
 
-def config
-def utils
-
-stage ('Checkout') {
-    node {
-        checkout scm
-        sh 'git submodule sync'
-        sh 'git submodule update --init --recursive'
-        // clean up
-        sh 'make clean'
-        // load the config
-        config = readYaml file: 'jenkins.yml'
-        // load the utility functions used below
-        utils = load 'jenkins/utils.groovy'
-        // defined in the Library loaded above
-        setGitEnvironmentVariables()
-        setConfigEnvironmentVariables(config)
-        // save the files for later
-        stash name: 'scripts', includes: 'bin/,docker/'
-        stash name: 'tests', includes: 'tests/,requirements/'
-        stash 'workspace'
+def loadBranch(String branch) {
+    // load the utility functions used below
+    utils = load 'jenkins/utils.groovy'
+    is_demo_branch = ( env.BRANCH_NAME ==~ /^demo\/[\w-]+$/ )
```

Comment:
I don't have any strong objections to breaking backwards compatibility, but I would recommend getting feedback from @alexgibson @jpetto and/or @craigcook as it would impact them.